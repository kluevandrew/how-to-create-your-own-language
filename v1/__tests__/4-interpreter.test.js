const { tokenize } = require('../1-tokenizer');
const { parse } = require('../2-parser');
const { compile } = require('../3-compiler');
const { Scope, evaluate } = require('../4-interpreter');

describe('evaluate', () => {
  it('should evaluate', () => {
    const log = [];
    const randomValue = 42;

    const scope = new Scope();

    scope.set('print', (arg) => {
      log.push(arg);
    });

    scope.set('a', () => {
      return randomValue;
    });

    evaluate(compile(parse(tokenize('(print (a))'))), scope);

    expect(log).toMatchSnapshot();
  });
});
