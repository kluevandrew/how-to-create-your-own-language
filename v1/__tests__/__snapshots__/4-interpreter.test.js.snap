// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`evaluate should evaluate 1`] = `
Array [
  42,
]
`;
